### ISSUE TITLE
How to upload array of files with params

### ISSUE THREAD

**Issue by selvam347** [gin-gonic/gin#774]
I can upload single file using [https://github.com/gin-gonic/gin/issues/548](url). 
```
package main
import "github.com/gin-gonic/gin"
import "log"
import "net/http"
import "io"
import "os"

func main() {
	router := gin.Default()
	router.POST("/upload", func(c *gin.Context) {
		file, handler, err := c.Request.FormFile("upload")
		filename := handler.Filename
		log.Println("Received file:", handler.Filename)
		out, err := os.Create("./tmp/" + filename)
		if err != nil {
			log.Fatal(err)
		}
		defer out.Close()
		_, err = io.Copy(out, file)
		if err != nil {
			log.Fatal(err)
		}
		c.String(http.StatusOK, "Uploaded...")
	})
	router.Run(":8080")
}
```
But how to upload array of files like the following,
`curl -X POST http://localhost:8080/upload -F "images[]=@/home/amp/hack/property1.jpg" -F "images[]=@/home/ack/property2.jpeg" -H "Content-Type: multipart/form-data"`

And give me suggestion for the same above code, how to get some parameters with documents. Finally request will be like,
```
{
  "client": {
    "name": "",
    "Industry": "",
    "profile_image": "PROFILE-IMAGE-FILE",
    "supporting_docs": {
      "docs": [
        "DOCUMENT1-FILE",
        "DOCUMENT2-FILE"
      ],
      "comments": "Comment test"
    }
  }
} 
```

**Comment by tboerger** [comment]
@andreynering does this also work with your form array value patch?

**Comment by andreynering** [comment]
Haven't tried by I don't think so.

`c.Request.FormFile` is Go's method, not Gin's.

**Comment by appleboy** [comment]
@tboerger @selvam347 @andreynering 

I create new PR to support upload single or multiple files. https://github.com/gin-gonic/gin/pull/775

```go
package main

import (
	"github.com/gin-gonic/gin"
	"log"
	"net/http"
)

func main() {
	router := gin.Default()
	router.POST("/upload", func(c *gin.Context) {
		// Multipart form
		form, _ := c.MultipartForm()
		files := form.File["upload[]"]

		for _, file := range files {
			log.Println(file.Filename)
		}
		c.String(http.StatusOK, "Uploaded...")
	})
	router.Run(":8080")
}
```

**Comment by appleboy** [comment]
Closed by #775

**Comment by roushanj** [comment]
How to upload In S3 Bucket ?
